### DISCUSSION TITLE
Check if user is logged in in pimcore

### DISCUSSION THREAD

**Discussion by leuchtdiode** [pimcore/pimcore#12617] [Q&A / Support]
Hi,

I want to show an edit button on my frontend pages where an editor can follow the documents deep link directly to the editmode (assumed that he is logged in in pimcore).

So I need a way in my application (best thing would be in a twig extension) to determine if the user/visitor is logged in in pimcore backend in his session. Is there a class which I can use?

Any help would be appreciated.

Thx & BR,
Alex

**Comment by dvesh3** [comment]
@leuchtdiode you can authenticate frontend user against admin session through pre auth authentication firewall in your `security.yaml` like mentioned [here](https://pimcore.com/docs/pimcore/current/Development_Documentation/Development_Tools_and_Details/Security_Authentication/Authenticate_Against_Admin_Session.html)

**Comment by leuchtdiode** [comment]
@dvesh3 Thank you. And how would I check in my twig extension if the user is logged in? When I try to add Symfony\Component\Security\Core\Security via constructor (dependency injection) symfony immediately throws "A Token was not found in the TokenStorage." when user is not logged in and I'm trying to load the frontend page.

EDIT: I added "anonymous: true" to firewall and now it seems to work. Is this the correct way to do it? I'm not that Symfony approved ;-)

**Reply by dvesh3** [reply]
you can authenticate a user with `isGranted('ROLE_USER')` https://github.com/pimcore/demo/blob/10.x/src/Controller/AccountController.php#L77

**Reply by leuchtdiode** [reply]
> you can authenticate a user with `isGranted('ROLE_USER')` https://github.com/pimcore/demo/blob/10.x/src/Controller/AccountController.php#L77

Unfortunately `isGranted('ROLE_USER')` is always `false` in my document Action though I'm logged in in pimcore in this session?

I'm using this twig extension which seems to work when I set `anonymous: true` in security.yaml. Is this wrong?

```
<?php
namespace App\Twig\Extension;

use Pimcore\Model\Document;
use Symfony\Component\Security\Core\Security;
use Twig\Extension\AbstractExtension;
use Twig\TwigFunction;

class Admin extends AbstractExtension
{
	public function __construct(
		private Security $security
	)
	{
	}

	/**
	 * @return TwigFunction[]
	 */
	public function getFunctions(): array
	{
		return [
			new TwigFunction('admin', [ $this, 'admin' ]),
		];
	}

	public function admin(): self
	{
		return $this;
	}

	public function isLoggedIn(): bool
	{
		$user = $this->security->getUser();

		return isset($user);
	}

	public function getPageDeepLink(Document\Page $page): string
	{
		return '/admin/login/deeplink?document_' . $page->getId() . '_page';
	}
}
```

**Reply by leuchtdiode** [reply]
@dvesh3 Any news regarding my code snippet? :)

Thx & BR,
Alex

**Reply by dvesh3** [reply]
Sorry, please check with `AuthorizationChecker::isGranted('ROLE_PIMCORE_USER')`.

**Reply by leuchtdiode** [reply]
> ROLE_PIMCORE_USER

This is working, thank you :)

But just for "symfony learning reasons": Would my approach via twig extension be wrong or does it have any security flaws?

BR,
Alex

**Comment by leuchtdiode** [comment]
@dvesh3 Hi, i just updated pimcore from v10.6.9 to to v11.1.3 and my solution to check if a user is logged in in the frontend seems to be not working any longer. I had this in my `security.yaml` under `security.firewalls`:

```
cms_firewall:
    anonymous: true
    pimcore_admin_pre_auth: ~
```

But it looks like this is not a valid confuguration any longer.

When I take this config out, the `AuthorizationChecker::isGranted('ROLE_PIMCORE_USER')` always returns `false`.

Do you have any tips how I can get this working again?

Thx & BR,
Alex
